src/actions/importing/aioseo-cleanup-action.php

```diff
@@ -0,0 +1,217 @@
+		'_aioseo_og_description',
+		'_aioseo_twitter_title',
+		'_aioseo_twitter_description',
+	];
+
+	/**
+	 * The WordPress database instance.
+	 *
+	 * @var wpdb
+	 */
+	protected $wpdb;
+
+	/**
+	 * The wpdb helper.
+	 *
+	 * @var Wpdb_Helper
+	 */
+	protected $wpdb_helper;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param wpdb           $wpdb        The WordPress database instance.
+	 * @param Options_Helper $options     The options helper.
+	 * @param Wpdb_Helper    $wpdb_helper The wpdb_helper helper.
+	 */
+	public function __construct(
+		wpdb $wpdb,
+		Options_Helper $options,
+		Wpdb_Helper $wpdb_helper
+	) {
+		$this->wpdb        = $wpdb;
+		$this->options     = $options;
+		$this->wpdb_helper = $wpdb_helper;
+	}
+
+	/**
+	 * Retrieves the AIOSEO table name along with the db prefix.
+	 *
+	 * @return string The AIOSEO table name along with the db prefix.
+	 */
+	protected function get_aioseo_table() {
+		return $this->wpdb->prefix . 'aioseo_posts';
+	}
+
+	/**
+	 * Retrieves the postmeta along with the db prefix.
+	 *
+	 * @return string The postmeta table name along with the db prefix.
+	 */
+	protected function get_postmeta_table() {
+		return $this->wpdb->prefix . 'postmeta';
+	}
+
+	/**
+	 * Determines if the AIOSEO database table exists.
+	 *
+	 * @return bool True if the table is found.
+	 */
+	protected function aioseo_exists() {
+		return $this->wpdb_helper->table_exists( $this->get_aioseo_table() ) === true;
+	}
+
+	/**
+	 * Returns whether the AISOEO post importing action is enabled.
+	 *
+	 * @return bool True if the AISOEO post importing action is enabled.
+	 */
+	public function is_enabled() {
+		$aioseo_importer_conditional = \YoastSEO()->classes->get( AIOSEO_V4_Importer_Conditional::class );
+
+		return $aioseo_importer_conditional->is_met();
+	}
+
+	/**
+	 * Just checks if the cleanup has been completed in the past.
+	 *
+	 * @return int The total number of unimported objects.
+	 */
+	public function get_total_unindexed() {
+		if ( ! $this->aioseo_exists() ) {
+			return 0;
+		}
+
+		return ( ! $this->get_completed() ) ? 1 : 0;
+	}
+
+	/**
+	 * Just checks if the cleanup has been completed in the past.
+	 *
+	 * @param int $limit The maximum number of unimported objects to be returned.
+	 *
+	 * @return int|false The limited number of unindexed posts. False if the query fails.
+	 */
+	public function get_limited_unindexed_count( $limit ) {
+		if ( ! $this->aioseo_exists() ) {
+			return 0;
+		}
+
+		return ( ! $this->get_completed() ) ? 1 : 0;
+	}
+
+	/**
+	 * Cleans up AIOSEO data.
+	 *
+	 * @return Indexable[]|false An array of created indexables or false if aioseo data was not found.
+	 */
+	public function index() {
+		if ( $this->get_completed() ) {
+			return [];
+		}
+
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared -- Reason: There is no unescaped user input.
+		$meta_data  = $this->wpdb->query( $this->cleanup_postmeta_query() );
+		$indexables = $this->wpdb->query( $this->truncate_query() );
```

Comment:
`$indexables` feel a bit confusing given how often we use this elsewhere to indicate an array of indexables. Would rename it to something like `$aioseo_table_truncate_done`.